src/main/java/io/vertx/core/parsetools/JsonEvent.java

```diff
@@ -0,0 +1,169 @@
+   */
+  JsonEventType type();
+
+  /**
+   * @return the name of the field when the event is emitted as a JSON object member
+   */
+  String fieldName();
+
+  /**
+   * @return the json value for {@link JsonEventType#VALUE} events
+   */
+  Object value();
+
+  /**
+   * @return true when the JSON value is a number
+   */
+  boolean isNumber();
+
+  /**
+   * @return the {@code Integer} value or {@code null} if the event has no JSON value
+   * @throws java.lang.ClassCastException if the value is not an {@code Integer}
+   */
+  Integer integerValue();
+
+  /**
+   * @return the {@code Long} value or {@code null} if the event has no JSON value
+   * @throws java.lang.ClassCastException if the value is not a {@code Long}
+   */
+  Long longValue();
+
+  /**
+   * @return the {@code Float} value or {@code null} if the event has no JSON value
+   * @throws java.lang.ClassCastException if the value is not a {@code Float}
+   */
+  Float floatValue();
+
+  /**
+   * @return the {@code Double} value or {@code null} if the event has no JSON value
+   * @throws java.lang.ClassCastException if the value is not a {@code Double}
+   */
+  Double doubleValue();
+
+  /**
+   * @return true when the JSON value is a boolean
+   */
+  boolean isBoolean();
+
+  /**
+   * @return the {@code Boolean} value or {@code null} if the event has no JSON value
+   * @throws java.lang.ClassCastException if the value is not a {@code Boolean}
+   */
+  Boolean booleanValue();
+
+  /**
+   * @return true when the JSON value is a string
+   */
+  boolean isString();
+
+  /**
+   * @return the string value or {@code null} if the event has no JSON value
+   * @throws java.lang.ClassCastException if the value is not a string
+   */
+  String stringValue();
+
+  /**
+   * Return the binary value.
+   * <p>
+   * JSON itself has no notion of a binary, this extension complies to the RFC-7493, so this method assumes there is a
+   * String value with the key and it contains a Base64 encoded binary, which it decodes if found and returns.
+   *
+   * @return the binary value or {@code null} if the event has no JSON value
+   * @throws java.lang.ClassCastException if the value is not a String
+   * @throws java.lang.IllegalArgumentException if the String value is not a legal Base64 encoded value
+   */
+  byte[] binaryValue();
+
+  /**
+   * Return the {@code Instant} value.
+   * <p>
+   * JSON itself has no notion of a temporal types, this extension complies to the RFC-7493, so this method assumes
+   * there is a String value with the key and it contains an ISO 8601 encoded date and time format
+   * such as "2017-04-03T10:25:41Z", which it decodes if found and returns.
+   *
+   * @return the {@code Instant} value or {@code null} if the event has no JSON value
+   * @throws java.lang.ClassCastException if the value is not a String
+   * @throws java.time.format.DateTimeParseException if the String value is not a legal ISO 8601 encoded value
+   */
+  Instant instantValue();
```

Comment:
I thought `Instant` didn't work with codegen

Reply:
yes, I'm going to add "@GenIgnore", I was testing using the IDE that does not run the annotation processor but in reality it fails :-)